File: HospitalDuty.Domain/Entities/ShiftPreference.cs
using System;
using HospitalDuty.Domain.Enums;

namespace HospitalDuty.Domain.Entities;

public class ShiftPreference
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid EmployeeId { get; set; }
    public Employee Employee { get; set; }
    public DateTime Date { get; set; }
    public PreferenceType PreferenceType { get; set; } = PreferenceType.Unavailable;
    public string Notes { get; set; } = string.Empty;
    public string DepartmentName { get; set; } = string.Empty;
}

File: client/src/types/shiftPreference.ts
export interface ShiftPreference {
  id: string;
  employeeId: string;
  date: string; // ISO string
  preferenceType: PreferenceType;
  notes: string;
  departmentName: string;
  employeeName: string;
}

export interface CreateShiftPreferenceRequest {
  // employeeId artık otomatik olarak current user'dan alınıyor
  dates: Date[]; // Date array
  preferenceType: PreferenceType;
  notes: string;
}

export const PreferenceType = {
  Unavailable: 0,
  Preferred: 1
} as const;

export type PreferenceType = typeof PreferenceType[keyof typeof PreferenceType];

export interface ShiftPreferenceResponse {
  success: boolean;
  message: string;
  data?: ShiftPreference[];
}
